src/test/java/com/redhat/jenkins/plugins/ci/integration/AmqMessagingPluginIntegrationTest.java

```diff
@@ -421,6 +421,33 @@ public void testSimpleCIEventTriggerWithParamOverride() throws Exception, Interr
     }
 
     @Test
+    public void testSimpleCIEventTriggerHeadersInEnv() throws Exception, InterruptedException {
+        FreeStyleJob jobA = jenkins.jobs.create();
+        jobA.configure();
+        CIEventTrigger ciEvent = new CIEventTrigger(jobA);
+        ciEvent.selector.set("CI_TYPE = 'code-quality-checks-done'");
+
+        // We are only checking that this shows up in the console output.
+        jobA.addShellStep("echo $CI_HEADERS");
+        jobA.save();
+
+        FreeStyleJob jobB = jenkins.jobs.create();
+        jobB.configure();
+        CINotifierPostBuildStep notifier = jobB.addPublisher(CINotifierPostBuildStep.class);
+
+        notifier.messageType.select("CodeQualityChecksDone");
+        notifier.messageContent.set("some irrelevant content");
+
+        jobB.save();
+        jobB.startBuild().shouldSucceed();
+
+        elasticSleep(1000);
+        jobA.getLastBuild().shouldSucceed().shouldExist();
+        assertThat(jobA.getLastBuild().getConsole(), containsString("\"CI_STATUS\":\"passed\""));
+        assertThat(jobA.getLastBuild().getConsole(), containsString("\"CI_TYPE\":\"code-quality-checks-done\""));
```

Comment:
Can try.  I could parse it out of the console log, but that's ugly.  I'll try to see if there's a way in the test here to access the env vars directly.